### PR TITLE
Various changes

### PR DESCRIPTION
**Pull request by dani-garcia**
Hey, I've been using your library in one of my personal projects and it's been working great!

I've made a series of small changes that might be of interest to you, I'll include them all on this PR to avoid spamming, but I put them in separate commits for easier reading. If you want to merge only one part, I can create a separate individual PR.

A small note about the TXT change, the original way (using the append_qname) gave me wrong results when viewing them from a separate mdns discovery app from my phone, so I changed it to a simple string which made it work, but I can't vouch for it's correctness.

### CODE REVIEW

File: src/dns/packet.rs [hunk 1]
```diff
@@ -60,6 +61,13 @@ impl<'a> PacketBuilder<'a> {
         self
     }
 
+    /// Adds an answer to the packet with class IN
+    pub fn add_answer_in(&mut self, name: &'a str, ttl: Duration, data: RData<'a>) -> &mut Self {
+        self.answers.push(ResourceRecord { name, ttl, class: Class::IN, data });
+        self.header.an_count += 1;
+        self
+    }
+
```
Comment (MarinPostma) [inline src/dns/packet.rs]: I'm not so sure about this method, but I agree that the ergonomics could be better, I have something else in mind for this.

**Review by MarinPostma — CHANGES_REQUESTED** [review]
Hello @dani-garcia, thank you for being the first contributor to this project! As you see this is still a very WIP project, and your contribution is greatly appreciated.

I have a few concerns with the `add_answer_in` method for now. I'm still working on this API design, and I'd like to put this change aside for now.

File: src/dns/packet.rs (hunk 1 above)
Comment (dani-garcia) [reply src/dns/packet.rs]: Yeah can't say i was a big fan of it either, I've removed it for now and pushed the changes.

**Review by MarinPostma — APPROVED** [review]
The txt does not seem to be iso with the RFC, but it's working ok for now, we'll patch it in the future. Thanks for your contibution

**Pull request merged**